www/js/search-database.js

```diff
@@ -82,36 +65,41 @@ module.exports = {
             angSearchSourceId = global.core.search.currentMeta.source;
             break;
         }
-        condition = `${searchCol} = ${dbQuery} AND v.SourceID = '${angSearchSourceId}'`;
+        condition = `${searchCol} = ${dbQuery} AND Source.SourceID = '${angSearchSourceId}'`;
         break;
       default:
         break;
     }
-    order.push('s.ShabadID');
-    const query = `SELECT ${allColumns} WHERE ${condition} ORDER BY ${order.join()} LIMIT ${limit}`;
-    this.db.all(query, (err, rows) => {
-      global.core.search.printResults(rows);
-    });
+    order.push('Shabads');
+    condition = `${condition} SORT(${order.join(' ASC, ')} ASC)`;
+    Realm.open(realmDB.realmVerseSchema)
+      .then((realm) => {
+        const rows = realm.objects('Verse').filtered(condition);
+        global.core.search.printResults(rows.slice(0, 20));
+      });
   },
 
   loadShabad(ShabadID, LineID) {
-    global.platform.db.all(`SELECT v.ID, v.Gurmukhi, v.English, v.Transliteration, v.punjabiUni, v.SourceID, v.PageNo AS PageNo FROM Verse v LEFT JOIN Shabad s ON v.ID = s.VerseID WHERE s.ShabadID = '${ShabadID}' ORDER BY v.ID`, (err, rows) => {
-      if (rows.length > 0) {
-        global.core.search.printShabad(rows, ShabadID, LineID || rows[0].ID);
-      }
-    });
+    Realm.open(realmDB.realmVerseSchema)
+      .then((realm) => {
+        const rows = realm.objects('Verse').filtered('ANY Shabads.ShabadID == $0', ShabadID);
```

Comment:
Somewhere the `ShabadID` should be checked for a number. when `Daily Hukamnama` is clicked, I see this error:
```
...www/js/search-database.js:85 Uncaught (in promise) Error: Property must be of type 'number', got (2685)
    at Realm.open.then (/Users/tsingh7/development/sttm-desktop/www/js/search-database.js:85)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188)
```

this is the line in `menu.js`. Probably needs to be converted from a string to a number.
```
getJSON('https://api.banidb.com/hukamnama/today', (error, response) => {
          goToShabadPage(response.shabadinfo.id);
        });
```

Reply:
You are correct. Once fix-73 gets merged into this branch, I'll address